src/main/kotlin/io/libp2p/core/pubsub/PubsubApi.kt

```diff
@@ -6,11 +6,18 @@ import io.libp2p.pubsub.PubsubRouter
 import io.netty.buffer.ByteBuf
 import java.util.concurrent.CompletableFuture
 import java.util.function.Consumer
+import java.util.function.Function
 import kotlin.random.Random.Default.nextLong
 
 fun createPubsubApi(router: PubsubRouter): PubsubApi =
     PubsubApiImpl(router)
 
+typealias Subscriber = Consumer<MessageApi>
+typealias Validator = Function<MessageApi, CompletableFuture<Boolean>>
```

Comment:
@shahankhatch sorry couldn't understand your idea :( 
May be a code snippet for illustration would help?

Reply:
I've thought on this further, and I've a better understanding of the API. In particular, I see that `PubSubApiImpl` has a validator parameter which will have some form of subscription capability implemented within it. This alternate path doesn't use a `Consumer` which was what confused me, I think. Since I understand this now, everything seems fine.